### PR TITLE
Fix ore generator NullPointerException

### LINKED ISSUE #253: NullPointerException on BlockFormEvent
[20:47:27 ERROR]: Could not pass event BlockFormEvent to IridiumSkyblock v3.0.6
java.lang.NullPointerException: null
        at com.iridium.iridiumskyblock.listeners.BlockFormListener.onBlockForm(BlockFormListener.java:46) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor406.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.16.5.jar:git-Purpur-1051]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.16.5.jar:git-Purpur-1051]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.16.5.jar:git-Purpur-1051]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607) ~[patched_1.16.5.jar:git-Purpur-1051]
        at org.bukkit.craftbukkit.v1_16_R3.event.CraftEventFactory.handleBlockFormEvent(CraftEventFactory.java:1713) ~[patched_1.16.5.jar:git-Purpur-1051]
        at org.bukkit.craftbukkit.v1_16_R3.event.CraftEventFactory.handleBlockFormEvent(CraftEventFactory.java:1705) ~[patched_1.16.5.jar:git-Purpur-1051]
        at org.bukkit.craftbukkit.v1_16_R3.event.CraftEventFactory.handleBlockFormEvent(CraftEventFactory.java:1664) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.BlockFluids.a(BlockFluids.java:159) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.BlockFluids.onPlace(BlockFluids.java:103) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.BlockBase$BlockData.onPlace(BlockBase.java:666) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.Chunk.setType(Chunk.java:739) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.World.a(World.java:661) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.World.setTypeAndData(World.java:622) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.FluidTypeFlowing.a(FluidTypeFlowing.java:312) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.FluidTypeLava.a(FluidTypeLava.java:202) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.FluidTypeFlowing.a(FluidTypeFlowing.java:192) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.FluidTypeFlowing.a(FluidTypeFlowing.java:157) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.FluidTypeFlowing.a(FluidTypeFlowing.java:516) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.Fluid.a(Fluid.java:56) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.WorldServer.a(WorldServer.java:1468) ~[patched_1.16.5.jar:git-Purpur-1051]
        at com.destroystokyo.paper.server.ticklist.PaperTickList.tick(PaperTickList.java:309) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.TickListServer.b(TickListServer.java:59) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.WorldServer.doTick(WorldServer.java:1063) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1554) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:430) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1403) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1129) ~[patched_1.16.5.jar:git-Purpur-1051]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:291) ~[patched_1.16.5.jar:git-Purpur-1051]
        at java.lang.Thread.run(Thread.java:834) [?:?]

### PR DESCRIPTION
**Pull request by dlsf**
Fixes #253

### CODE REVIEW

File: src/main/java/com/iridium/iridiumskyblock/utils/RandomAccessList.java [hunk 1]
```diff
@@ -44,11 +45,11 @@ public void addAll(Map<E, Integer> possibilities) {
...
      *
      * @return A random element
-     * @throws IndexOutOfBoundsException If this list is empty
      */
-    public E nextElement() throws IndexOutOfBoundsException {
+    public Optional<E> nextElement() throws IndexOutOfBoundsException {
+        if (underlyingList.isEmpty()) return Optional.empty();
         int randomIndex = ThreadLocalRandom.current().nextInt(underlyingList.size());
-        return underlyingList.get(randomIndex);
+        return Optional.of(underlyingList.get(randomIndex));
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumskyblock/utils/RandomAccessList.java:52]: Needs to be Optional.ofNullable cuz Optionals are dum

**Review by PeachesMLG — CHANGES_REQUESTED** [review]

File: src/main/java/com/iridium/iridiumskyblock/utils/RandomAccessList.java (hunk 1 above)
Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/utils/RandomAccessList.java:52]: It can't be null tho?

Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/utils/RandomAccessList.java:52]: The maximum of the index is always `underlyingList.size() - 1` and if the list is empty there is always an empty optional returned

Comment (PeachesMLG) [reply src/main/java/com/iridium/iridiumskyblock/utils/RandomAccessList.java:52]: then why is it an optional if Optional.isPresent is always gonna be true?

Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/utils/RandomAccessList.java:52]: It's not, check line 50

Comment (PeachesMLG) [reply src/main/java/com/iridium/iridiumskyblock/utils/RandomAccessList.java:52]: nevermind im blind

**Pull request merged**